src/Nethermind/Nethermind.Store/StateProvider.cs

```diff
@@ -179,6 +174,18 @@ private void SetNewBalance(Address address, in UInt256 balanceChange, IReleaseSp
             PushUpdate(address, changedAccount);
         }
 
+        private Account GetThroughCacheCheckExists(Address address)
```

Comment:
the merhod name does not suggest anything about updsting balance and the exception talks about updating balance